### PR TITLE
softmax configurable now

### PR DESCRIPTION
**Pull request by jjoshua2**

### CODE REVIEW

**Comment by killerducky** [comment]
We discussed this in #dev a bit: I don't think this is the correct temperature to tune. What we want to modify is how UCTNode::randomize_first_propotionally does. There isn't anything in that routine called temperature because it only supports T=1 right now. T=0 is what happens when cfg_randomize=false, and this routine is not called at all. To support other T values we need to do some sort of `pow` function to the accum var in rand_first_prop.

cfg_softmax_temp deals with the shape of the probability_p vector, which the search uses to decide which moves to search. Tuning this is hard because it's in the self-play training feedback loop, and would take extremely long experiments to test. Plus the google paper doesn't refer to this (for this part they just mention "logit", which to mean means to default version of logit aka softmax).
